### ISSUE TITLE
Spurious Fiber Failure Warning when ZIO.repeat is used

### ISSUE THREAD

**Issue by reibitto** [zio/zio#2309]
I get a fiber failure warning for this when it completes:
```scala
val schedule = Schedule.recurs(10) && Schedule.fixed(100.millis)
putStrLn("test").repeat(schedule)
```

Output:
```
A checked error was not handled.
()

Fiber:Id(1574499603310,35) was supposed to continue to: <empty trace>

Fiber:Id(1574499603310,35) execution trace:
  at zio.Schedule.check(Schedule.scala:238)
...
```

But this completes fine without any warnings:
```scala
val schedule = Schedule.recurs(10)
putStrLn("test").repeat(schedule)
```

Possibly related to #1949.

**Comment by reibitto** [comment]
It looks like it's the introduction of `&&` that causes this, not `Schedule.fixed` itself. For example, you can also see it with something like `Schedule.once && Schedule.once`. A plain `Schedule.once` is fine though.

**Comment by Twizty** [comment]
Hi, I'd like to implement it.

**Comment by reibitto** [comment]
@Twizty I haven't started this beyond looking for a minimal example. Sure, you can take it. Thanks!

**Comment by Twizty** [comment]
@reibitto ok, then I start working on it.

**Comment by thiloplanz** [comment]
Is this also the cause of the issue reported at https://stackoverflow.com/q/59063170/14955 ?

**Comment by adamgfraser** [comment]
@thiloplanz I think it is related but slightly different. We have a general problem where fiber failures are reported when sometimes they shouldn't be because they are "intentional" in some sense (e.g. `race` is supposed to interrupt the loser). This issue in particular is caused by `race` because `Schedule#&&` is implemented in terms of `zipPar` which is implemented in terms of `race`. The Stack Overflow example involves `orElse` which doesn't use `race` and raises a slightly different issue of a fiber that should not initially be silent but should be silent after `orElse` is called on it. I will open a ticket for it.

**Comment by adamgfraser** [comment]
Closed by #2339.
